internal/resources/slo/resource_slo.go

```diff
@@ -0,0 +1,412 @@
+package slo
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+
+	gapi "github.com/grafana/grafana-api-golang-client"
+
+	"github.com/grafana/terraform-provider-grafana/internal/common"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+func ResourceSlo() *schema.Resource {
+	return &schema.Resource{
+		Description: `
+Resource manages Grafana SLOs. 
+
+* [Official documentation](https://grafana.com/docs/grafana-cloud/slo/)
+* [API documentation](https://grafana.com/docs/grafana-cloud/slo/api/)
+		`,
+		CreateContext: resourceSloCreate,
+		ReadContext:   resourceSloRead,
+		UpdateContext: resourceSloUpdate,
+		DeleteContext: resourceSloDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:         schema.TypeString,
+				Required:     true,
+				Description:  `Name should be a short description of your indicator. Consider names like "API Availability"`,
+				ValidateFunc: validation.StringLenBetween(0, 128),
+			},
+			"description": &schema.Schema{
+				Type:         schema.TypeString,
+				Required:     true,
+				Description:  `Description is a free-text field that can provide more context to an SLO.`,
+				ValidateFunc: validation.StringLenBetween(0, 1024),
+			},
+			"query": &schema.Schema{
+				Type:        schema.TypeList,
+				Required:    true,
+				Description: `Query describes the indicator that will be measured against the objective. Freeform Query types are currently supported.`,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"type": &schema.Schema{
+							Type:         schema.TypeString,
+							Description:  `Query type must be one of: "freeform", "query", "ratio", or "threshold"`,
+							ValidateFunc: validation.StringInSlice([]string{"freeform", "query", "ratio", "threshold"}, false),
+							Required:     true,
+						},
+						"freeformquery": &schema.Schema{
```

Comment:
The API value is a struct with a single field. Since that's extensible in the future (maybe some metadata for dashboarding could be added?), l think we should mimic the structure here of a Freeform struct and a Query field.